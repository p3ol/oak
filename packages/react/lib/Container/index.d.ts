import { ReactNode, ComponentPropsWithoutRef } from 'react';
import {
  ElementObject,
  ComponentObject,
  Component,
} from '@oakjs/core';

export declare interface ContainerProps extends ComponentPropsWithoutRef<any> {
  element?: ElementObject;
  content?: Array<ElementObject>;
  component?: ComponentObject | Component;
  depth?: number;
}

declare function Container(props: ContainerProps): ReactNode | JSX.Element;

export default Container;
