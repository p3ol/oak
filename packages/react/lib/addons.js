import {
  TextField,
  SelectField,
  ColorField,
  DateField,
  Toggle,
} from '@junipero/react';
import { slideInDownMenu } from '@junipero/transitions';
import * as coreAddons from '@oakjs/core/addons';

import { dragOption, backgroundColorOption } from './options';
import { ImageField } from './fields';
import {
  Button,
  Col,
  EmptySpace,
  Foldable,
  Image,
  Row,
  Text as TextComponent,
  Title,
} from './components';
import Text from './Text';

export const textField = props => ({
  ...coreAddons.textField(),
  render: (fieldProps, { setting, t } = {}) => (
    <TextField
      { ...fieldProps }
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      type={setting.valueType || 'text'}
    />
  ),
  ...props,
});

export const textareaField = props => ({
  ...coreAddons.textareaField(),
  render: (fieldProps, { setting, t } = {}) => (
    <TextField
      { ...fieldProps }
      tag="textarea"
      rows={fieldProps.rows || 5}
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      type={setting.valueType || 'text'}
    />
  ),
  ...props,
});

export const selectField = props => ({
  ...coreAddons.selectField(),
  render: (fieldProps, { setting, editableRef, t } = {}) => (
    <SelectField
      clearable={false}
      { ...fieldProps }
      options={setting.options}
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      animateMenu={slideInDownMenu}
      parseTitle={setting.parseTitle || (o => o?.title ? t(o.title) : o)}
      parseValue={setting.parseValue || (o => o?.value ?? o)}
      container={editableRef.current}
    />
  ),
  ...props,
});

export const colorField = props => ({
  ...coreAddons.colorField(),
  render: (fieldProps, { setting, t } = {}) => (
    <ColorField
      { ...fieldProps }
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
    />
  ),
  ...props,
});

export const imageField = props => ({
  ...coreAddons.imageField(),
  render: (fieldProps, { setting, element }) => (
    <ImageField
      { ...fieldProps }
      iconOnly={setting?.props?.iconOnly}
      setting={setting}
      element={element}
    />
  ),
  ...props,
});

export const dateField = props => ({
  ...coreAddons.dateField(),
  render: (fieldProps, { setting, t } = {}) => (
    <DateField
      { ...fieldProps }
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
    />
  ),
  ...props,
});

export const toggleField = props => ({
  ...coreAddons.toggleField(),
  render: (fieldProps, { setting } = {}) => (
    <Toggle { ...fieldProps } checked={fieldProps.value}>
      <Text>
        { fieldProps.value
          ? setting.checkedLabel
          : setting.uncheckedLabel }
      </Text>
    </Toggle>
  ),
  ...props,
});

export const rowComponent = props => ({
  ...coreAddons.rowComponent(),
  render: Row,
  options: [dragOption(), backgroundColorOption()],
  ...props,
});

export const colComponent = props => ({
  ...coreAddons.colComponent(),
  render: Col,
  ...props,
});

export const emptySpaceComponent = props => ({
  ...coreAddons.emptySpaceComponent(),
  render: EmptySpace,
  ...props,
});

export const titleComponent = props => ({
  ...coreAddons.titleComponent(),
  render: Title,
  ...props,
});

export const textComponent = props => ({
  ...coreAddons.textComponent(),
  render: TextComponent,
  ...props,
});

export const imageComponent = props => ({
  ...coreAddons.imageComponent(),
  render: Image,
  ...props,
});

export const buttonComponent = props => ({
  ...coreAddons.buttonComponent(),
  render: Button,
  ...props,
});

export const foldableComponent = props => ({
  ...coreAddons.foldableComponent(),
  options: [dragOption()],
  render: Foldable,
  ...props,
});

export const baseFields = () => [
  textField(),
  textareaField(),
  selectField(),
  colorField(),
  imageField(),
  dateField(),
  toggleField(),
];

export const baseComponents = () => [
  rowComponent(),
  colComponent(),
  emptySpaceComponent(),
  titleComponent(),
  textComponent(),
  imageComponent(),
  buttonComponent(),
  foldableComponent(),
];

export const coreComponentsGroup = () => ({
  ...coreAddons.coreComponentsGroup(),
  components: baseComponents(),
});

export const baseAddon = () => ({
  ...coreAddons.baseAddon(),
  components: [coreComponentsGroup()],
  fields: baseFields(),
});
