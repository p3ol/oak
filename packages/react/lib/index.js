export {
  TextField,
  SelectField,
  DateField,
  ColorField,
  Card,
  Dropdown,
  DropdownMenu,
  DropdownToggle,
  Toggle,
  TouchableZone,
  Label,
  classNames,
  useTimeout,
} from '@junipero/react';

export { default as Builder } from './Builder';
export { default as Container } from './Container';
export { default as Catalogue } from './Catalogue';
export { default as Icon } from './Icon';
export { default as Text } from './Text';
export { default as Field } from './Editable/Field';
export { default as DisplayableSettings } from './DisplayableSettings';

export {
  useBuilder,
  useLogger,
} from './hooks';

export * from './fields';
export * from './options';
export * from './addons';
