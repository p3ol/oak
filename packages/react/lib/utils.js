export const copyToClipboard = value =>
  globalThis.navigator.clipboard.writeText(value);

export const sanitizeHTML = content => {
  try {
    const parsed = new DOMParser().parseFromString(content, 'text/html');

    // Remove script & style tags
    parsed
      .querySelectorAll('script, style, iframe, object, video, audio')
      .forEach(item => item.parentNode.removeChild(item));

    // Disable all links
    parsed.querySelectorAll('a[href]').forEach(item => {
      item.removeAttribute('href');
      item.removeAttribute('target');
      item.setAttribute('role', 'link');
    });

    return parsed.body.innerHTML;
  } catch (e) {
    console.error(e);

    return '';
  }
};
