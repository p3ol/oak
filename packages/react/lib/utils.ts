export const copyToClipboard = (value: string): Promise<void> =>
  globalThis.navigator.clipboard.writeText(value);

export const sanitizeHTML = (content: string): string => {
  try {
    const parsed = new DOMParser().parseFromString(content, 'text/html');

    // Remove script & style tags
    parsed
      .querySelectorAll('script, style, iframe, object, video, audio')
      .forEach(item => item.parentNode.removeChild(item));

    // Disable all links
    parsed.querySelectorAll('a[href]').forEach(item => {
      item.removeAttribute('href');
      item.removeAttribute('target');
      item.setAttribute('role', 'link');
    });

    return parsed.body.innerHTML;
  } catch (e) {
    console.error(e);

    return '';
  }
};
