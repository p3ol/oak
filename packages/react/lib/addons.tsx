import type { ComponentsGroupObject, AddonObject } from '@oakjs/core';
import {
  type TextFieldProps,
  type SelectFieldProps,
  type ColorFieldProps,
  type ToggleProps,
  type DateFieldProps,
  TextField,
  SelectField,
  ColorField,
  DateField,
  Toggle,
  classNames,
} from '@junipero/react';
import { slideInDownMenu } from '@junipero/transitions';
import * as coreAddons from '@oakjs/core/addons';

import type { ReactComponentObject, ReactFieldObject } from './types';
import { dragOption, backgroundColorOption } from './options';
import { type ImageFieldProps, ImageField } from './fields';
import {
  Button,
  Col,
  EmptySpace,
  Foldable,
  Image,
  Row,
  Text as TextComponent,
  Title,
} from './components';
import Text from './Text';

export const textField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.textField(),
  render: (fieldProps: TextFieldProps, { setting, t } = {}) => (
    <TextField
      { ...fieldProps }
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      type={setting.valueType || 'text'}
    />
  ),
  ...props,
});

export const textareaField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.textareaField(),
  render: (fieldProps: TextFieldProps, { setting, t } = {}) => (
    <TextField
      { ...fieldProps }
      tag="textarea"
      rows={fieldProps.rows || 5}
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      type={setting.valueType || 'text'}
    />
  ),
  ...props,
});

export const selectField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.selectField(),
  render: (fieldProps: SelectFieldProps, { setting, editableRef, t } = {}) => (
    <SelectField
      clearable={false}
      { ...fieldProps }
      options={setting.options}
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      animateMenu={slideInDownMenu}
      parseTitle={setting.parseTitle || (o => o?.title ? t(o.title) : o)}
      parseValue={setting.parseValue || (o => o?.value ?? o)}
      container={editableRef.current?.innerRef.current as HTMLDivElement}
    />
  ),
  ...props,
});

export const colorField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.colorField(),
  render: (fieldProps: ColorFieldProps, { setting, editableRef, t } = {}) => (
    <ColorField
      { ...fieldProps }
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      container={editableRef.current?.innerRef.current as HTMLDivElement}
    />
  ),
  ...props,
});

export const imageField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.imageField(),
  render: (fieldProps: ImageFieldProps, { setting, element }) => (
    <ImageField
      { ...fieldProps }
      className={classNames(fieldProps?.className, setting?.props?.className)}
      iconOnly={setting?.props?.iconOnly}
      setting={setting}
      element={element}
    />
  ),
  ...props,
});

export const dateField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.dateField(),
  render: (fieldProps: DateFieldProps, { setting, editableRef, t } = {}) => (
    <DateField
      { ...fieldProps }
      { ...setting.placeholder && {
        placeholder: t(setting.placeholder),
      } }
      container={editableRef.current?.innerRef.current as HTMLDivElement}
    />
  ),
  ...props,
});

export const toggleField = (props?: ReactFieldObject): ReactFieldObject => ({
  ...coreAddons.toggleField(),
  render: (fieldProps: ToggleProps, { setting } = {}) => (
    <Toggle { ...fieldProps } checked={fieldProps.value}>
      <Text>
        { fieldProps.value
          ? setting.checkedLabel
          : setting.uncheckedLabel }
      </Text>
    </Toggle>
  ),
  ...props,
});

export const rowComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.rowComponent(),
  render: Row,
  options: [dragOption(), backgroundColorOption()],
  ...props,
});

export const colComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.colComponent(),
  render: Col,
  ...props,
});

export const emptySpaceComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.emptySpaceComponent(),
  render: EmptySpace,
  ...props,
});

export const titleComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.titleComponent(),
  render: Title,
  ...props,
});

export const textComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.textComponent(),
  render: TextComponent,
  ...props,
});

export const imageComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.imageComponent(),
  render: Image,
  ...props,
});

export const buttonComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.buttonComponent(),
  render: Button,
  ...props,
});

export const foldableComponent = (
  props?: ReactComponentObject
): ReactComponentObject => ({
  ...coreAddons.foldableComponent(),
  options: [dragOption()],
  render: Foldable,
  ...props,
});

export const baseFields = (): ReactFieldObject[] => [
  textField(),
  textareaField(),
  selectField(),
  colorField(),
  imageField(),
  dateField(),
  toggleField(),
];

export const baseComponents = (): ReactComponentObject[] => [
  rowComponent(),
  colComponent(),
  emptySpaceComponent(),
  titleComponent(),
  textComponent(),
  imageComponent(),
  buttonComponent(),
  foldableComponent(),
];

export const coreComponentsGroup = (): ComponentsGroupObject => ({
  ...coreAddons.coreComponentsGroup(),
  components: baseComponents(),
});

export const baseAddon = (): AddonObject => ({
  ...coreAddons.baseAddon(),
  components: [coreComponentsGroup()],
  fields: baseFields(),
  overrides: [{
    type: 'setting',
    targets: ['*'],
    key: 'styles.backgroundImage',
    props: {
      iconOnly: true,
      className: 'oak-mr-4 oak-relative oak-top-[2px]',
    },
  }],
});
