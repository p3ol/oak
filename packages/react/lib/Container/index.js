import { useRef } from 'react';
import { Droppable, classNames } from '@junipero/react';

import { useBuilder } from '../hooks';
import Element from '../Element';
import Catalogue from '../Catalogue';

const Container = ({
  element,
  component,
  className,
  content = [],
  depth = 0,
}) => {
  const prependCatalogueRef = useRef();
  const appendCatalogueRef = useRef();
  const { builder } = useBuilder();

  const onPrepend = c => {
    prependCatalogueRef.current?.close();
    builder.addElement?.({}, {
      parent: content,
      position: 'before',
      component: c,
    });
  };

  const onAppend = c => {
    appendCatalogueRef.current?.close();
    builder.addElement?.({}, {
      parent: content,
      position: 'after',
      component: c,
    });
  };

  const onDrop = data => {
    if (component?.disallow?.includes?.(data.type)) {
      return;
    }

    builder.moveElement?.(data, element, {
      parent: content,
      position: 'after',
    });
  };

  const onPasteBefore = elmt => {
    prependCatalogueRef.current?.close();
    builder.addElements([].concat(elmt || []), {
      parent: content,
      position: 'before',
      resetIds: true,
    });
  };

  const onPasteAfter = elmt => {
    appendCatalogueRef.current?.close();
    builder.addElements([].concat(elmt || []), {
      parent: content,
      position: 'after',
      resetIds: true,
    });
  };

  return (
    <Droppable disabled={content.length > 0} onDrop={onDrop}>
      <div
        className={classNames(
          'container oak-flex-auto oak-flex oak-flex-col oak-gap-2',
          'oak-p-4',
          depth % 2 === 0 ? 'even' : 'odd',
          className
        )}
        data-depth={depth}
      >
        { content.length > 0 && (
          <Catalogue
            ref={prependCatalogueRef}
            component={component}
            onAppend={onPrepend}
            onPaste={onPasteBefore}
            className="oak-inline-flex oak-self-center small"
          />
        ) }

        { content.length > 0 && (
          <div className="content oak-flex oak-flex-col oak-gap-4">
            { content.map((elt, i) => (
              <Element
                depth={depth + 1}
                key={elt.id || i}
                index={i}
                element={elt}
                parent={content}
                parentComponent={component}
              />
            )) }
          </div>
        ) }

        <Catalogue
          ref={appendCatalogueRef}
          component={component}
          onAppend={onAppend}
          onPaste={onPasteAfter}
          className={classNames(
            'oak-inline-flex oak-self-center',
            { small: content.length > 0 }
          )}
        />
      </div>
    </Droppable>
  );
};

Container.displayName = 'Container';

export default Container;
