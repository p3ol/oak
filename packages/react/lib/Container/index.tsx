import type { Component, ComponentObject, ElementObject } from '@oakjs/core';
import {
  type ComponentPropsWithoutRef,
  useCallback,
  useMemo,
  useRef,
} from 'react';
import { Droppable, classNames } from '@junipero/react';

import { useBuilder } from '../hooks';
import Element from '../Element';
import Catalogue, { type CatalogueRef } from '../Catalogue';

export declare interface ContainerProps
  extends Omit<ComponentPropsWithoutRef<'div'>, 'content'> {
  element?: ElementObject;
  content?: Array<ElementObject>;
  component?: ComponentObject | Component;
  depth?: number;
  droppable?: boolean;
}

const Container = ({
  element,
  component,
  className,
  droppable = true,
  content = [],
  depth = 0,
  ...rest
}: ContainerProps) => {
  const prependCatalogueRef = useRef<CatalogueRef>(null);
  const appendCatalogueRef = useRef<CatalogueRef>(null);
  const { builder, addons } = useBuilder();
  const override = useMemo(() => (
    builder.getOverride('component', component?.id) as ComponentObject
  ), [component, builder, addons]);

  const onPrepend = (c: Component | ComponentObject) => {
    prependCatalogueRef.current?.close();
    builder.addElement?.({}, {
      parent: content,
      position: 'before',
      component: c,
    });
  };

  const onAppend = (c: Component | ComponentObject) => {
    appendCatalogueRef.current?.close();
    builder.addElement?.({}, {
      parent: content,
      position: 'after',
      component: c,
    });
  };

  const onDrop = useCallback((
    data: ElementObject,
  ) => {
    if (
      component?.disallow?.includes?.(data.type) ||
      override?.disallow?.includes?.(data.type)
    ) {
      return;
    }

    builder.moveElement?.(data, element, {
      parent: content,
      position: 'after',
    });
  }, [builder, content, element, component, override]);

  const onPasteBefore = (elmt: ElementObject) => {
    prependCatalogueRef.current?.close();
    builder.addElements([].concat(elmt || []), {
      parent: content,
      position: 'before',
      resetIds: true,
    });
  };

  const onPasteAfter = (elmt: ElementObject) => {
    appendCatalogueRef.current?.close();
    builder.addElements([].concat(elmt || []), {
      parent: content,
      position: 'after',
      resetIds: true,
    });
  };

  return (
    <Droppable
      disabled={
        content.length > 0 ||
        (override?.droppable ?? droppable) === false
      }
      onDrop={onDrop}
    >
      <div
        { ...rest }
        className={classNames(
          'container oak-flex-auto oak-flex oak-flex-col oak-gap-2',
          'oak-p-4',
          depth % 2 === 0 ? 'even' : 'odd',
          className
        )}
        data-depth={depth}
      >
        { content.length > 0 && (
          <Catalogue
            ref={prependCatalogueRef}
            component={component}
            onAppend={onPrepend}
            onPaste={onPasteBefore}
            className="oak-inline-flex oak-self-center small"
          />
        ) }

        { content.length > 0 && (
          <div className="content oak-flex oak-flex-col oak-gap-4">
            { content.map((elt, i) => (
              <Element
                depth={depth + 1}
                key={elt.id || i}
                index={i}
                element={elt}
                parent={content}
                parentComponent={component}
              />
            )) }
          </div>
        ) }

        <Catalogue
          ref={appendCatalogueRef}
          component={component}
          onAppend={onAppend}
          onPaste={onPasteAfter}
          className={classNames(
            'oak-inline-flex oak-self-center',
            { small: content.length > 0 }
          )}
        />
      </div>
    </Droppable>
  );
};

Container.displayName = 'Container';

export default Container;
