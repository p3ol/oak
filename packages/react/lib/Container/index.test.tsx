import type { BuilderObject, ElementObject } from '@oakjs/core';
import { fireEvent, render, within } from '@testing-library/react';

import { BuilderLite } from '../../tests/utils';
import { baseAddon } from '../addons';
import Container from './index';

describe('<Container />', () => {
  const getOptions = (props?: BuilderObject) => {
    let i = 0;

    return {
      generateId: () => i++,
      ...props,
    };
  };

  it('should render', () => {
    const elmt: ElementObject = {
      id: '1',
      type: 'foldable',
      content: [],
    };

    const { container, unmount } = render(
      <BuilderLite options={getOptions()} addons={[baseAddon()]}>
        <Container
          element={elmt}
          content={elmt.content as ElementObject[]}
        />
      </BuilderLite>
    );
    expect(container).toMatchSnapshot();

    // Open catalogue
    const catalogue = container.querySelector<HTMLElement>('.catalogue');
    fireEvent.click(within(catalogue).getByText('add'));

    // Add a text
    fireEvent.click(within(container).getByText('Text'));

    expect(elmt).toMatchSnapshot('With element added');
    unmount();
  });

  it('should allow to prepend elements inside a container', () => {
    const elmt = { id: '1', type: 'foldable', content: [
      { id: '2', type: 'text', content: 'This is a text' },
    ] };

    const { container, unmount } = render(
      <BuilderLite options={getOptions()} addons={[baseAddon()]}>
        <Container
          element={elmt}
          content={elmt.content}
        />
      </BuilderLite>
    );

    // Open catalogue
    const catalogue = container
      .querySelector<HTMLElement>('.catalogue:first-child');
    fireEvent.click(within(catalogue).getByText('add'));

    // Add a text
    const menu = container.querySelector<HTMLElement>('.catalogue-menu');
    fireEvent.click(within(menu).getByText('Text'));

    expect(elmt).toMatchSnapshot('With element added before');
    unmount();
  });
});
