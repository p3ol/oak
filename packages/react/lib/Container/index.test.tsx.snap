// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should allow to prepend elements inside a container: With element added before 1`] = `
{
  "content": [
    {
      "content": "This is some fancy text content",
      "id": 0,
      "type": "text",
    },
    {
      "content": "This is a text",
      "id": "2",
      "type": "text",
    },
  ],
  "id": "1",
  "type": "foldable",
}
`;

exports[`<Container /> should render 1`] = `
<div>
  <div>
    <div
      class="container oak-flex-auto oak-flex oak-flex-col oak-gap-2 oak-p-4 even"
      data-depth="0"
    >
      <div
        class="catalogue oak-inline-flex oak-self-center"
      >
        <a
          class="handle oak-inline-flex oak-items-center oak-justify-center"
          draggable="false"
        >
          <i
            class="icon junipero-icons"
          >
            add
          </i>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Container /> should render: With element added 1`] = `
{
  "content": [
    {
      "content": "This is some fancy text content",
      "id": 0,
      "type": "text",
    },
  ],
  "id": "1",
  "type": "foldable",
}
`;
